internal/server/server.go

```diff
@@ -855,6 +857,8 @@ func (server *Server) handleInputCommand(client *Client, msg *Message) error {
 			return writeErr("read only")
 		}
 	case "eval", "evalsha":
+		write = true // We are going assume here that any eval is a write operation on this layer
```

Comment:
I don't understand this.  We're not going to replay 'EVAL FOO' command in the replica, therefore we don't need a `write=true` here.